### DISCUSSION TITLE
Inequality in jump_commands.s test

### DISCUSSION THREAD

**Discussion by gilyaakov** [amichai-bd/rvc_asap#27] [General]
![Inequality](https://user-images.githubusercontent.com/72501420/159310927-2bab5056-abd9-4253-8fc4-cf95712100bf.jpg)

As can be seen, we get a mismatch between the expected image and the real image of the jump test. We have researched this in depth and this is not a problem in our architecture. The problem stems from the following differences.

In the Venus software from which we extract the expected memory image of the test, the following commands exist: 

![image](https://user-images.githubusercontent.com/72501420/159311660-a7e6ca78-99ff-428e-8834-6eb3ab174449.png)

While in the elf.txt file (of the test) these 2 commands exist as one command:

![image](https://user-images.githubusercontent.com/72501420/159312029-f00feefd-57e7-4f41-ba17-c0a51a1923a2.png)

As a result, the following differences were created:

**In the real image**

![image](https://user-images.githubusercontent.com/72501420/159312330-205afb59-7c60-4289-a676-cc57c7ebc49e.png)

**In the golden image**

![image](https://user-images.githubusercontent.com/72501420/159312463-eb8f6867-11c8-4b0d-bf78-849358cd8dd1.png)

Hence the difference between 104 and 108.

**Comment by amichai-bd** [comment]
This is an interesting case.
Might be resolved by using some flag in the compile/link command to remove optimizations.
In any case, if you confirmed your HW is executing the I_MEM Instructions correctly, you may fix the "golden" snapshot to match the correct expected code.
